game-app/game-core/src/main/java/games/strategy/triplea/ui/ProductionRepairPanel.java

```diff
@@ -269,7 +269,8 @@ private ResourceCollection getResources() {
       final int bid = data.getProperties().get(propertyName, 0);
       final ResourceCollection bidCollection = new ResourceCollection(data);
       try (GameData.Unlocker ignored = data.acquireReadLock()) {
-        bidCollection.addResource(data.getResourceList().getResource(Constants.PUS), bid);
+        bidCollection.addResource(
```

Comment:
@BBlaza Not all marked from above, but same here: Either the method assumes the value must be available, then new method `getResourceOrThrow?` or a default value should be used, e.g., `orElse(0)`.